### PR TITLE
feature: Add support for third level navigation

### PR DESCRIPTION
**Pull request by luilliarcec**

### CODE REVIEW

File: packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
+@props([
+    'collapsible' => true,
+    'label' => null,
+    'name' => null,
+    'icon' => null,
+])
+
+<li x-data="{ label: {{ \Illuminate\Support\Js::from($name) }} }"
+    @class(['filament-sidebar-subgroup overflow-hidden'])
+>
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php]:
```suggestion
<li
    x-data="{ label: @js($name) }"
    class="filament-sidebar-subgroup overflow-hidden"
>
```

File: packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                'dark:text-gray-300' => config('filament.dark_mode'),
+            ])>
+                {{ $label }}
+            </div>
+
+            @if ($collapsible)
+                <x-heroicon-o-chevron-down :class="\Illuminate\Support\Arr::toCssClasses([
+                    'w-3 h-3 text-gray-600 transition',
+                    'dark:text-gray-300' => config('filament.dark_mode'),
+                ])" x-bind:class="$store.sidebar.groupIsCollapsed(label) || '-rotate-180'" x-cloak/>
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php]:
```suggestion
                <x-heroicon-o-chevron-down
                    :class="\Illuminate\Support\Arr::toCssClasses([
                        'w-3 h-3 text-gray-600 transition',
                        'dark:text-gray-300' => config('filament.dark_mode'),
                    ])"
                    x-bind:class="$store.sidebar.groupIsCollapsed(label) || '-rotate-180'"
                    x-cloak
                />
```

File: packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            @endif
+        </button>
+    @endif
+
+    <ul
+        x-show="! ($store.sidebar.groupIsCollapsed(label) && {{ config('filament.layout.sidebar.is_collapsible_on_desktop') ? '$store.sidebar.isOpen' : 'true' }})"
```
Comment (zepfietje) [inline packages/admin/resources/views/components/layouts/app/sidebar/subgroup.blade.php]:
```suggestion
        x-show="!($store.sidebar.groupIsCollapsed(label) && {{ config('filament.layout.sidebar.is_collapsible_on_desktop') ? '$store.sidebar.isOpen' : 'true' }})"
```

File: packages/admin/src/FilamentManager.php [hunk 4]
```diff
@@ -231,7 +233,19 @@ public function getNavigation(): array
...
                 return NavigationGroup::make($registeredGroup ?? $groupIndex)
-                    ->items($items);
+                    ->subgroups(
+                        $items
+                            ->filter(fn (Navigation\NavigationItem $item): bool => $item->getSubgroup() !== null)
+                            ->groupBy(fn (NavigationItem $item): string => $item->getSubgroup())
+                            ->map(
+                                fn (Collection $items, string $subgroup): NavigationSubgroup => NavigationSubgroup::make($subgroup)
+                                    ->icon($items->first()->getSubgroupIcon())
+                                    ->items($items)
```
Comment (zepfietje) [inline packages/admin/src/FilamentManager.php]:
```suggestion
                                    ->items($items),
```

File: packages/admin/src/FilamentManager.php [hunk 5]
```diff
@@ -231,7 +233,19 @@ public function getNavigation(): array
...
+                            )
+                    )
+                    ->items(
+                        $items->filter(fn (Navigation\NavigationItem $item): bool => $item->getSubgroup() === null)
```
Comment (zepfietje) [inline packages/admin/src/FilamentManager.php]:
```suggestion
                        $items->filter(fn (Navigation\NavigationItem $item): bool => $item->getSubgroup() === null),
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

File: packages/admin/src/FilamentManager.php [hunk 6]
```diff
@@ -231,7 +233,19 @@ public function getNavigation(): array
...
-                    ->items($items);
+                    ->subgroups(
+                        $items
+                            ->filter(fn (Navigation\NavigationItem $item): bool => $item->getSubgroup() !== null)
+                            ->groupBy(fn (NavigationItem $item): string => $item->getSubgroup())
+                            ->map(
+                                fn (Collection $items, string $subgroup): NavigationSubgroup => NavigationSubgroup::make($subgroup)
+                                    ->icon($items->first()->getSubgroupIcon())
+                                    ->items($items),
+                            )
```
Comment (zepfietje) [inline packages/admin/src/FilamentManager.php]:
```suggestion
                            ),
```

**Review by zepfietje — CHANGES_REQUESTED** [review]

**Comment by danharrin** [comment]
Hey, there is alot of complexity here because we're allowing resources and pages to register subgroups. Personally I think this should only be an option for the [Advanced Navigation API](https://filamentphp.com/docs/2.x/admin/navigation#advanced-navigation-customization). Also, please add docs so I know how to use it.

**Comment by Nuhel** [comment]
> Hey, there is alot of complexity here because we're allowing resources and pages to register subgroups. Personally I think this should only be an option for the [Advanced Navigation API](https://filamentphp.com/docs/2.x/admin/navigation#advanced-navigation-customization). Also, please add docs so I know how to use it.

I think it wil be nice, If there is a class NavigationGrouo like NavigationItem class, And NavigationGrouo allow both NavigationItem and NavigationGroup as child. When there is NavigationGroup as child, It will be rendered as a group.

**Comment by danharrin** [comment]
yeah I agree with @Nuhel, introducing a "subgroup" is not optimal when we could just support nested groups

**Comment by Nuhel** [comment]
I have added support for multilevel menu, [You can have a look](https://drive.google.com/file/d/1mKdPiEoCZ1RThOOZ_WeBTof6eGx7V5OG/view?usp=sharing).

If you guys like this I will make an pr.

**Comment by danharrin** [comment]
I don't care about the design, we can fix that. If the backend implementation is not good enough then it will not be merged.

**Comment by Nuhel** [comment]
> I don't care about the design, we can fix that. If the backend implementation is not good enough then it will not be merged.

Yes I know, I just wanted to be sure, If the design is very bad , i would try to fix the design and than make an PR.

**Comment by danharrin** [comment]
Closing in favour of #4650

**Pull request closed without merging**